src/ServicePulse.Host/app/js/views/pending_retries/controller.js

```diff
@@ -0,0 +1,246 @@
+            }
+
+            row.selected = !row.selected;
+
+            if (row.selected) {
+                vm.selectedIds.push(row.id);
+            } else {
+                vm.selectedIds.splice(vm.selectedIds.indexOf(row.id), 1);
+            }
+        }
+
+        vm.toggleRowSelect = function (row) {
+            toggleSelect(row);
+
+            vm.allSelected = vm.selectedIds.length === vm.pendingRetryMessages.length;
+        };
+
+        vm.retrySelected = function () {
+            serviceControlService.retryFailedMessages(vm.selectedIds);
+            vm.selectedIds = [];
+
+            for (var i = 0; i < vm.pendingRetryMessages.length; i++) {
+                if (vm.pendingRetryMessages[i].selected) {
+                    vm.pendingRetryMessages[i].selected = false;
+                    vm.pendingRetryMessages[i].retried = true;
+                }
+            }
+        };
+
+        vm.markAsResolvedSelected = function () {
+            serviceControlService.markAsResolvedMessages(vm.selectedIds);
+            vm.selectedIds = [];
+
+            for (var i = 0; i < vm.pendingRetryMessages.length; i++) {
+                if (vm.pendingRetryMessages[i].selected) {
+                    vm.pendingRetryMessages[i].selected = false;
+                    vm.pendingRetryMessages[i].resolved = true;
+                }
+            }
+        };
+
+        vm.archiveSelected = function () {
+            serviceControlService.archiveFailedMessages(vm.selectedIds);
+            vm.selectedIds = [];
+
+            for (var i = 0; i < vm.pendingRetryMessages.length; i++) {
+                if (vm.pendingRetryMessages[i].selected) {
+                    vm.pendingRetryMessages[i].selected = false;
+                    vm.pendingRetryMessages[i].archived = true;
+                }
+            }
+        };
+
+        vm.searchPhraseChanged = function() {
+            if (!vm.searchPhrase) {
+                vm.pendingRetryMessages = [];
+                vm.page = 1;
+                vm.loadMoreResults();
+            }
+        }
+
+        vm.onSelect = function () {
+            vm.pendingRetryMessages = [];
+            vm.page = 1;
+            vm.loadMoreResults();
+        }
+
+        vm.debugInServiceInsight = function (index) {
+            var messageId = vm.pendingRetryMessages[index].message_id;
+            var dnsName = scConfig.service_control_url.toLowerCase();
+
+            if (dnsName.indexOf("https") === 0) {
+                dnsName = dnsName.replace("https://", "");
+            } else {
+                dnsName = dnsName.replace("http://", "");
+            }
+
+            $window.open("si://" + dnsName + "?search=" + messageId);
+        };
+
+        var selectGroupInternal = function (sort, direction, changeToMessagesTab) {
+            vm.sort = sort;
+            vm.direction = direction;
+            setSortButtonText(sort, direction);
+
+            if ($scope.loadingData) {
+                return;
```

Comment:
Do we want the sort button text to be set if the code breaks out without actually loading the results? i.e. in the case where `$scope.loadingData` is true